pkg/util/iterator.go

```diff
@@ -0,0 +1,142 @@
+package util
```

Comment:
Add a comment that this is copied from the prometheus code, and the PR to make it public?